core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -1864,35 +2221,36 @@ private int addColumnToMeta(
         return index;
     }
 
-    private void attachPartitionCheckFilesMatchFixedColumn(FilesFacade ff, Path path, int columnType, long partitionSize, String columnName, long columnNameTxn) {
-        TableUtils.dFile(path, columnName, columnNameTxn);
-        if (ff.exists(path.$())) {
-            long fileSize = ff.length(path);
+    private void attachPartitionCheckFilesMatchFixedColumn(int columnType, long partitionSize, String columnName, long columnNameTxn) {
+        TableUtils.dFile(detachedPath, columnName, columnNameTxn);
+        if (ff.exists(detachedPath.$())) {
+            long fileSize = ff.length(detachedPath);
             if (fileSize < partitionSize << ColumnType.pow2SizeOf(columnType)) {
                 throw CairoException.instance(0)
                         .put("Column file is too small. ")
                         .put("Partition files inconsistent [file=")
-                        .put(path)
+                        .put(detachedPath)
                         .put(", expectedSize=")
                         .put(partitionSize << ColumnType.pow2SizeOf(columnType))
                         .put(", actual=")
                         .put(fileSize)
                         .put(']');
             }
-            return;
         }
-        throw CairoException.instance(0).put("Column file does not exist [path=").put(path).put(']');
     }
 
-    private void attachPartitionCheckFilesMatchMetadata(FilesFacade ff, Path path, RecordMetadata metadata, long partitionSize) throws CairoException {
+    private void attachPartitionCheckFilesMatchMetadata(long partitionSize) throws CairoException {
         // for each column, check that file exists in the partition folder
-        int rootLen = path.length();
-        for (int columnIndex = 0, size = metadata.getColumnCount(); columnIndex < size; columnIndex++) {
+        int rootLen = detachedPath.length();
+        for (int i = 0, size = metadata.getColumnCount(); i < size; i++) {
             try {
-                int columnType = metadata.getColumnType(columnIndex);
+                int columnIndex = metadata.getWriterIndex(i);
                 final String columnName = metadata.getColumnName(columnIndex);
+                if (detachedAddMissingColNames.contains(columnName)) {
+                    continue;
+                }
+                int columnType = metadata.getColumnType(columnIndex);
                 long columnNameTxn = columnVersionWriter.getDefaultColumnNameTxn(columnIndex);
```

Comment:
should be `detachedColumnVersionWriter` if the detached cv file is in the folder